go-controller/pkg/ovn/pods.go

```diff
@@ -381,5 +386,46 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	// observe the pod creation latency metric.
 	recordPodCreated(pod)
 
+	if err = oc.addLogicalSwitchPodCount(logicalSwitchInfo, logicalSwitch); err != nil {
+		return fmt.Errorf("failed to update pod count for the logical switch %s for pod %s - %v",
+			logicalSwitch, pod.Name, err)
+	}
+
+	return nil
+}
+
+// addLogicalSwitchPodCount updates the per node metric for ip utilization.
+// It also increments the podCount cache for the switch given by nodeName.
+func (oc *Controller) addLogicalSwitchPodCount(lsInfo *logicalSwitchInfo, nodeName string) error {
+	if lsInfo == nil {
+		return fmt.Errorf("Invalid logical switch info while adding pod count for node %s", nodeName)
+	}
+	if nodeName == "" {
+		return fmt.Errorf("Invalid node name provided while incrementing pod count")
+	}
+	lsInfo.podCount = lsInfo.podCount + 1
+	updateNodeIPPoolUtilization(lsInfo.podCount, nodeName)
+	oc.lsMutex.Lock()
+	defer oc.lsMutex.Unlock()
+	oc.logicalSwitchCache[nodeName] = *lsInfo
+	return nil
+}
+
+// deleteLogicalSwitchPodCount updates the per node metric for ip utilization.
+// It also decrements the podCount cache for the switch given by nodeName by one.
+func (oc *Controller) deleteLogicalSwitchPodCount(nodeName string) error {
+	if nodeName == "" {
+		return fmt.Errorf("Invalid node name provided while decrementing pod count")
+	}
```

Comment:
nodeName will not be `empty` here, so perhaps you don't need the check above.

Also, I don't think this function should return an error. It should just log Warning and return.